src/routing/authorisedRoute.component.tsx

```diff
@@ -57,8 +60,16 @@ export default function withAuth<T>(
               }}
             />
           ) : null}
+          {/* If using a plugin as the start page, redirect here so the plugin renders with the redirected url */}
           {!loading && loggedIn ? (
-            <ComponentToProtect {...(componentProps as T)} />
+            startUrlState && startUrlState.scigateway.startUrl ? (
+              <Redirect
+                push
+                to={{ pathname: startUrlState.scigateway.startUrl }}
+              />
+            ) : (
```

Comment:
Why is this bit needed? If I comment this out it still redirects to the plugin with the correct URL?

Reply:
When I tested this, navigating to `/` while logged in redirects to the right url (e.g. `/browse/investigation`) and it loads the plugin, but the page that's rendered isn't the page that matches the current url, but the previous url (`/`). So for this example, it displays the "Browse investigations" link, as that's what the plugin routing renders: https://github.com/ral-facilities/datagateway/blob/master/packages/datagateway-dataview/src/page/pageTable.component.tsx#L41
Adding in this redirect caused it to re-render and show the table as expected. As I recall, search and dataview display links in this way, but I don't know what download does off the top of my head.
If this isn't an issue on your end for whatever reason, and the page renders as expected rather than showing that link, then there wouldn't be any reason to include this redirect.

Reply:
Ah, I was testing this with just the demo plugin rather than one of the datagateway plugins - after testing with dataview you're indeed correct. I assumed it was needed for a reason - thanks for the explanation :)

Reply:
No problem, I could remember there was something that happened differently depending on whether the user was logged in or not when I first saw the comment, but it took me a bit of poking around to remind myself exactly what the problem was.